pkg/policy/policies_compute.go

```diff
@@ -0,0 +1,142 @@
+package policy
+
+import (
+	"github.com/aquasecurity/tracee/pkg/filters"
+	"github.com/aquasecurity/tracee/pkg/utils"
+)
+
+// compute recalculates values, updates flags, fills the reduced userland map,
+// and sets the related bitmap that is used to prevent the iteration of the entire map.
+// It also computes the events states.
+//
+// It must be called at initialization and at every runtime policies changes.
+func (ps *Policies) compute() {
```

Comment:
I don't get it. Could you elaborate?

Reply:
So this function is responsible to updating all the filters of the `Policies` type.
You could make all of them their own struct. Keep a reference to it in the `Policies` struct. Copying it upon the call to `compute`. Updating the copy. In the end doing atomic swap to the pointer to the struct to point to the modified copy.
This way you don't need to concern yourself with mutexes for this operation, as the whole data is always consistent.

Reply:
Yea you are right, if the entire policy is read-only since initialization then its ok.
Do what you think is the cleanest code-wise.

Reply:
> I agree with you concerning inner structures to hold each step computed values, and returning them to single assignments. Let me work on that.

I'm going to tackle this in the sequential PR to avoid conflicts between then.